lib/cn/node_split.c

```diff
@@ -518,33 +528,54 @@ cn_split(struct cn_compaction_work *w)
 
     assert(!list_empty(&w->cw_node->tn_kvset_list));
 
+    wargs = calloc(w->cw_kvset_cnt, sizeof(*wargs));
+    if (!wargs)
+        return merr(ENOMEM);
+
+    wq = cn_get_io_wq(w->cw_tree->cn);
+
     for (i = 0, le = list_first_entry(&w->cw_node->tn_kvset_list, typeof(*le), le_link);
          i < w->cw_kvset_cnt;
          i++, le = list_next_entry(le, le_link)) {
 
-        struct kvset *ks = le->le_kvset;
-        struct kvset_split_res result = { 0 };
+        INIT_WORK(&wargs[i].work, kvset_split_worker);
+        wargs[i].ks = le->le_kvset;
+        wargs[i].split_kobj = &split_kobj;
+        wargs[i].pc = w->cw_pc;
+        kvset_split_res_init(w, &wargs[i].result, i);
+
+        if (!queue_work(wq, &wargs[i].work)) {
```

Comment:
What would cause `queue_work` to fail?

Reply:
queue_work fails if the work is already in the queue.. That should never happen and hence EBUG.